connect-inject/handler.go

```
@@ -504,6 +548,173 @@ func (h *Handler) defaultAnnotations(pod *corev1.Pod, patches *[]jsonpatch.JsonP
+// that with the annotation if provided.
+func (h *Handler) prometheusScrapePort(pod *corev1.Pod) (string, error) {
+	defaultPrometheusScrapePort := h.DefaultPrometheusScrapePort
+
+	return determineAndValidatePort(pod, annotationPrometheusScrapePort, defaultPrometheusScrapePort)
+}
+
+// prometheusScrapePath returns the default value in the handler, or overrides
+// that with the annotation if provided.
+func (h *Handler) prometheusScrapePath(pod *corev1.Pod) (string, error) {
+	defaultPrometheusScrapePath := h.DefaultPrometheusScrapePath
+
+	if raw, ok := pod.Annotations[annotationPrometheusScrapePath]; ok && raw != "" {
+		return raw, nil
+	}
+
+	return defaultPrometheusScrapePath, nil
+}
+
+// serviceMetricsPort returns the port used to register the service with Consul,
+// or overrides that with the annotation if provided.
+func (h *Handler) serviceMetricsPort(pod *corev1.Pod) (string, error) {
+	// The annotationPort is the port used to register the service with Consul.
+	// If that has been set, it'll be used as the port for getting service
+	// metrics as well, unless overridden by the service-metrics-port annotation.
+	if raw, ok := pod.Annotations[annotationPort]; ok && raw != "" {
+		return determineAndValidatePort(pod, annotationServiceMetricsPort, raw)
+	}
+
+	// If the annotationPort is not set, the serviceMetrics port will be 0
+	// unless overridden by the service-metrics-port annotation. If the service
+	// metrics port is 0, the consul sidecar will not run a merged metrics
+	// server.
+	return determineAndValidatePort(pod, annotationServiceMetricsPort, "0")
+}
+
+// serviceMetricsPath returns a default of /metrics, or overrides
+// that with the annotation if provided.
+func (h *Handler) serviceMetricsPath(pod *corev1.Pod) (string, error) {
+	defaultServiceMetricsPath := "/metrics"
+
+	if raw, ok := pod.Annotations[annotationServiceMetricsPath]; ok && raw != "" {
+		return raw, nil
+	}
+
+	return defaultServiceMetricsPath, nil
```

You could potentially have a file level const `const string defaultServiceMetricsPath = "/metrics"`
```suggestion
	if raw, ok := pod.Annotations[annotationServiceMetricsPath]; ok && raw != "" {
		return raw, nil
	}
	return "/metrics", nil
```

---

Also, do we need to return an error here? It doesnt look like it ever gets set.

---

Changed it to use the const, and got rid of the error, thanks!!